### ISSUE TITLE
Results saving issues on train and detect

### ISSUE THREAD

**Issue by julesmorata** [ultralytics/yolov5#10118]
Hey,
I encounter some issues with results saving with yolov5 used with command lines from a python script.
I can't figure out how to generate the 'exp' folder when doing so, it seems to work when executing it as a python library (import + detect.run()) but it does not with the following line :

os.system('cmd /c "python yolov5/train.py --data path_to_config/config.yaml --weights yolov5l.pt --epochs {} --batch {} --freeze 10"'.format(nb_epochs_seg, batch_size))

or 

os.system('cmd /c "python detect.py --source {} --weights {} --save-txt'.format(os.path.join(image_path, 'test'), os.path.join(os.path.join('yolov5/runs/train', os.listdir('yolov5/runs/train')[-1]), 'weights/best.pt')))

I managed to get through it for the inference as I use this : detect.run(source=os.path.join(image_path, 'test'), save_txt=True) but I can't do it for the training because it leads to an error about memory allocated to pytorch.

And another problem that I am facing is that when saving detection results I only get a label file when at least an object is detected, I believe this is the normal behaviour but is there an option to also generate empty files when nothing gets detected ?

Thanks.

**Comment by glenn-jocher** [comment]
@bugrahanismailoglu hi there,

It looks like the 'exp' folder isn't being generated because you're invoking YOLOv5 via `os.system` in your Python script. The 'exp' folder is typically auto-generated when running YOLOv5 directly. You might encounter this issue due to memory allocation for PyTorch during training.

Regarding your second query, by default, YOLOv5 saves labels only for detected objects. Generating empty files when nothing is detected typically isn't the default behavior.

If you'd like to discuss this further or need help with anything else, please feel free to reach out.

Best regards.
